src/builder/create_allowed_mentions.rs

```diff
@@ -73,11 +73,13 @@ impl CreateAllowedMentions {
     /// [`roles`]: Self::roles
     #[inline]
     pub fn parse(&mut self, value: ParseValue) -> &mut Self {
-        if let Some(val) = self.0.get_mut("parse") {
-            val.as_array_mut().unwrap().push(json![value]);
-        } else {
-            self.0.insert("parse", Value::Array(vec![json!(value)]));
+        let val = self.0.entry("parse").or_insert_with(|| Value::Array(Vec::new()));
+
+        if let Some(arr) = val.as_array_mut() {
+            arr.push(json![value]);
         }
+        // Should there be an "else" block that panics?
```

Comment:
It's a programmatic error for the `parse` field to not be an array, though it's very unlikely to happen. The field is managed by these builder methods, which ensure the field is an array. The only way it could be another type is if the user inserted the field themselves, since the `HashMap` is public. For the sake of being pedantic, I say there should be a panic.

```rust
let arr = val.as_array_mut().expect("must be an array");
arr.push(json![value]);
```